Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,106 @@
+              [this](RobotImporter::LogLevel level, const AZStd::string& message)
+              {
+                  switch (level)
+                  {
+                  case RobotImporter::LogLevel::Info:
+                      ReportInfo(message);
+                      break;
+                  case RobotImporter::LogLevel::Error:
+                      ReportError(message);
+                      break;
+                  }
+              })
+    {
+        setWindowTitle(QObject::tr("Robot definition file importer"));
+        QVBoxLayout* mainLayout = new QVBoxLayout(this);
+        mainLayout->setSpacing(20);
+        QLabel* captionLabel = new QLabel(QObject::tr("Select a Unified Robot Description Format (URDF) file to import"), this);
+        captionLabel->setWordWrap(true);
+        mainLayout->addWidget(captionLabel);
+        mainLayout->addWidget(&m_selectFileButton);
+        mainLayout->addWidget(&m_statusLabel);
+        mainLayout->addStretch();
+
+        connect(
+            &m_importerUpdateTimer,
+            &QTimer::timeout,
+            [this]
+            {
+                m_robotImporter.CheckIfAssetsWereLoadedAndCreatePrefab(
+                    [this]()
+                    {
+                        m_importerUpdateTimer.stop();
+                        m_selectFileButton.setEnabled(true);
+                    });
+            });
+
+        QObject::connect(
+            &m_selectFileButton,
+            &QPushButton::clicked,
+            this,
+            [this]()
+            {
+                AZStd::optional<AZStd::string> urdfPath = RobotImporterWidgetUtils::QueryUserForURDFPath(this);
+                if (!urdfPath)
+                {
+                    return;
+                }
+
+                AZ::IO::Path prefabName(AZ::IO::PathView(urdfPath.value()).Filename());
+                prefabName.ReplaceExtension("prefab");
+                AZStd::string prefabDefaultPath(AZ::IO::Path(AZ::Utils::GetProjectPath()) / "Assets" / "Importer" / prefabName);
+                auto prefabPath =
+                    RobotImporterWidgetUtils::ValidatePrefabPathExistenceAndQueryUserForNewIfNecessary(prefabDefaultPath, this);
+                if (!prefabPath)
+                {
+                    ReportError("User cancelled");
+                    return;
+                }
+                m_robotImporter.ParseURDFAndStartLoadingAssets({ urdfPath.value(), prefabPath->c_str() });
+
+                // Disable the button until the import is complete to prevent the user from clicking it again
+                m_selectFileButton.setEnabled(false);
+                // Check whether import is still in progress every 0.5 seconds
+                m_importerUpdateTimer.start(500);
+            });
+        setLayout(mainLayout);
+    }
+
+    void RobotImporterWidget::ReportError(const AZStd::string& errorMessage)
+    {
+        QMessageBox::critical(this, QObject::tr("Error"), QObject::tr(errorMessage.c_str()));
```

Comment:
Is it a good idea to trigger a message box here during an import? What if there was a problem that affected a lot of assets and triggers an error. This message box could pop up a lot of times, forcing the operator to click ok many times.

Reply:
Multiple errors during import are only logged into the console/log file.

`ReportError` function is intended to be used as a final message to the user (that the import failed and why). It should not be called repeatedly within the importer pipeline.

Reply:
As @piotr-zyskowski-rai mentioned, this box only handles the end result. However..
URDF importer in current version has very little UX.
Considering that it is already a huge PR, I propose to handle UX in a dedicated issue: https://github.com/RobotecAI/o3de-ros2-gem/issues/163.
Let me know if this is ok.

Reply:
As long as this is triggered at the end of the entire process, then its fine.